functions/index.js

```diff
@@ -4,36 +4,49 @@ const { authenticateApiKey, parseEmailFromRequest } = require("./mailing/middlew
 const { addToMailingList } = require("./mailing/mailchimp");
 
 exports.subscribe = functions.https.onRequest(async (request, response) => {
-  // TODO: refactor tests with firebase changes
-
-  const isAuthenticated = authenticateApiKey(functions, request);
-  const requestEmail = parseEmailFromRequest(request);
-
-  if (isAuthenticated === false) {
-    response.status(401).send({
-      error: true,
-      status: 401,
-      message: "Unable to authenticate request",
-    });
-  } else if (requestEmail === null) {
-    response.status(400).send({
-      error: true,
-      status: 400,
-      message: "Invalid or missing email in request body",
-    });
+  response.set("Access-Control-Allow-Origin", "https://www.cruzhacks.com");
+  if (request.method === "OPTIONS") {
+    response.set("Access-Control-Allow-Methods", "POST");
+    response.set("Access-Control-Allow-Headers", "authentication, Content-Type");
+    response.status(200).send('');
```

Comment:
Since it sends a preflight request, we have to handle it accordingly by setting the correct `Access-Control-Allow` headers. This will allow for the actual request to come through and execute properly.